app/controllers/AuditController.scala

```diff
@@ -64,9 +65,12 @@ class AuditController @Inject() (implicit val env: Environment[User, SessionAuth
           UserCurrentRegionTable.assignNextRegion(user.userId)
           region = RegionTable.selectTheCurrentNamedRegion(user.userId)
         }
+        println("Current assignment: " + region.get.regionId)
 
-        val task: NewTask = if (region.isDefined) AuditTaskTable.selectANewTaskInARegion(region.get.regionId, user)
-                            else AuditTaskTable.selectANewTask(user.username)
+        val task: NewTask = if (region.isDefined) AuditTaskTable.selectANewTaskInARegion(region.get.regionId, user.userId)
+                            else AuditTaskTable.selectANewTask(user.userId)
+        region = RegionTable.selectTheCurrentNamedRegion(user.userId)
```

Comment:
Was this line added for testing purposes? Because we already select a task in the previous line in the selected region.